File: sample/src/main/java/com/sscl/basesample/activities/sample/AnimationSampleActivity.java
package com.sscl.basesample.activities.sample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.animation.AccelerateDecelerateInterpolator;
import android.view.animation.AnimationSet;
import android.view.animation.Interpolator;
import android.view.animation.ScaleAnimation;
import android.view.animation.TranslateAnimation;

import com.sscl.basesample.R;

/**
 * 动画测试
 *
 * @author pengh
 */
public class AnimationSampleActivity extends AppCompatActivity {

    private View animationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_animation_sample);
        animationView = findViewById(R.id.iv);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        animationView = null;
    }

    /**
     * 位移动画
     *
     * @param view View
     */
    public void translate(View view) {
        TranslateAnimation animation = new TranslateAnimation(0, 100, 0, 200);
        animation.setRepeatCount(0);
        animation.setFillAfter(true);
        animation.setDuration(3000);
        animationView.startAnimation(animation);
    }

    /**
     * 缩放动画
     *
     * @param view View
     */
    public void scale(View view) {
        ScaleAnimation animation = new ScaleAnimation(1, 0, 1, 0);
        animation.setRepeatCount(0);
        animation.setFillAfter(true);
        animation.setDuration(3000);
        animationView.startAnimation(animation);
    }

    /**
     * 缩放并位移
     * @param view View
     */
    public void scaleTranslate(View view) {
        TranslateAnimation translateAnimation = new TranslateAnimation(0, 100, 0, 200);
        ScaleAnimation scaleAnimation = new ScaleAnimation(1, 0, 1, 0);
        AnimationSet animationSet = new AnimationSet(true);
        animationSet.addAnimation(scaleAnimation);
        animationSet.addAnimation(translateAnimation);
        animationSet.setRepeatCount(0);
        animationSet.setDuration(3000);
        animationSet.setFillEnabled(true);
        animationSet.setFillAfter(true);
        animationSet.setInterpolator(new AccelerateDecelerateInterpolator());
        animationView.startAnimation(animationSet);

    }
}

File: sample/src/main/java/com/sscl/basesample/activities/sample/AnimationSampleActivity.kt
package com.sscl.basesample.activities.sample

import com.sscl.basesample.R
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.TranslateAnimation
import android.view.animation.ScaleAnimation
import android.view.animation.AnimationSet
import android.view.animation.AccelerateDecelerateInterpolator
import android.view.View

/**
 * 动画测试
 *
 * @author pengh
 */
class AnimationSampleActivity : AppCompatActivity() {
    private var animationView: View? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.com_sscl_basesample_activity_animation_sample)
        animationView = findViewById(R.id.iv)
    }

    override fun onDestroy() {
        super.onDestroy()
        animationView = null
    }

    /**
     * 位移动画
     *
     * @param view View
     */
    fun translate(view: View?) {
        val animation = TranslateAnimation(0f, 100f, 0f, 200f)
        animation.repeatCount = 0
        animation.fillAfter = true
        animation.duration = 3000
        animationView!!.startAnimation(animation)
    }

    /**
     * 缩放动画
     *
     * @param view View
     */
    fun scale(view: View?) {
        val animation = ScaleAnimation(1f, 0f, 1f, 0f)
        animation.repeatCount = 0
        animation.fillAfter = true
        animation.duration = 3000
        animationView!!.startAnimation(animation)
    }

    /**
     * 缩放并位移
     * @param view View
     */
    fun scaleTranslate(view: View?) {
        val translateAnimation = TranslateAnimation(0f, 100f, 0f, 200f)
        val scaleAnimation = ScaleAnimation(1f, 0f, 1f, 0f)
        val animationSet = AnimationSet(true)
        animationSet.addAnimation(scaleAnimation)
        animationSet.addAnimation(translateAnimation)
        animationSet.repeatCount = 0
        animationSet.duration = 3000
        animationSet.isFillEnabled = true
        animationSet.fillAfter = true
        animationSet.interpolator = AccelerateDecelerateInterpolator()
        animationView!!.startAnimation(animationSet)
    }
}